src/backend/distributed/commands/dependencies.c

```diff
@@ -224,15 +225,50 @@ GetDependencyCreateDDLCommands(const ObjectAddress *dependency)
 	{
 		case OCLASS_CLASS:
 		{
+			char relKind = get_rel_relkind(dependency->objectId);
+
 			/*
 			 * types have an intermediate dependency on a relation (aka class), so we do
 			 * support classes when the relkind is composite
 			 */
-			if (get_rel_relkind(dependency->objectId) == RELKIND_COMPOSITE_TYPE)
+			if (relKind == RELKIND_COMPOSITE_TYPE)
 			{
 				return NIL;
 			}
 
+			if (relKind == RELKIND_RELATION || relKind == RELKIND_PARTITIONED_TABLE ||
+				relKind == RELKIND_FOREIGN_TABLE)
+			{
+				Oid relationId = dependency->objectId;
+				if (IsCitusTable(relationId) && !IsTableOwnedByExtension(relationId))
+				{
+					/* skip table metadata creation when the Citus table is owned by an extension */
+					List *commandList = NIL;
+					List *tableDDLCommands = GetFullTableCreationCommands(relationId,
+																		  WORKER_NEXTVAL_SEQUENCE_DEFAULTS);
+
+					TableDDLCommand *tableDDLCommand = NULL;
+					foreach_ptr(tableDDLCommand, tableDDLCommands)
+					{
+						Assert(CitusIsA(tableDDLCommand, TableDDLCommand));
+						commandList = lappend(commandList, GetTableDDLCommand(
+												  tableDDLCommand));
+					}
+
+					List *sequenceDependencyCommandList = SequenceDependencyCommandList(
+						dependency->objectId);
+					commandList = list_concat(commandList, sequenceDependencyCommandList);
+
+					return commandList;
+				}
+			}
+
+			if (relKind == RELKIND_SEQUENCE)
```

Comment:
I'm probably missing something but couldn't follow easily.

Why we generate DDLs for sequences for both when we get here and when the table has a sequence dependency? Would that cause generating the same sequence multiple times?

Reply:
We are creating sequences using that function. With the above one we create the dependency on the worker node as we create sequence and tables separately.